pkg/artifactory/resource/repository/federated/federated.go

```diff
@@ -113,3 +128,63 @@ func PackMembers(members []Member, d *schema.ResourceData) error {
 
 	return nil
 }
+func deleteRepo(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// For federated repositories we delete all the federated members, if the flag `cleanup_on_delete` is set to `true`
+	s := &util.ResourceData{ResourceData: d}
+	if s.GetBool("cleanup_on_delete", false) {
+		// Save base URL from the Client to be able to revert it back after the change below
+		baseURL := m.(util.ProvderMetadata).Client.BaseURL
+		if v, ok := d.GetOk("member"); ok {
+			federatedMembers := v.(*schema.Set).List()
+			if len(federatedMembers) == 0 {
+				return nil
```

Comment:
This will short circuit the rest of the delete request. This would work if this entire `if` block is in its own function.

Reply:
Removed, and also added verification to skip delete if the member is the initial repo.